### PR TITLE
Spec: Add FuncAttrLive

### PR DESCRIPTION
**Pull request by kubo39**
See also https://github.com/dlang/dmd/pull/11049

### CODE REVIEW

**Comment by PetarKirov** [comment]
Can you also add `Nk` for return parameters (https://github.com/dlang/druntime/commit/87fbb2885462f6a869161aa581efa21231d693d3)? 

BTW, I wonder why there is need for both `Parameter` and `Parameter2`. Instead it would be simpler if we defined it as:

```
Parameter:
    Type
    J Parameter     // out
    K Parameter     // ref
    L Parameter     // lazy
    M Parameter   // scope
    Nk Parameter  // return
```

@ibuclaw @WalterBright what do you think?

File: spec/abi.dd [hunk 1]
```diff
@@ -581,6 +585,7 @@ $(GNAME Parameter2):
     $(B J) $(GLINK Type)     $(GREEN // out)
     $(B K) $(GLINK Type)     $(GREEN // ref)
     $(B L) $(GLINK Type)     $(GREEN // lazy)
+    $(B Nk) $(GLINK Type)     $(GREEN // return)
```
Comment (PetarKirov) [inline spec/abi.dd:588]: I think that if we keep the current split between `Parameter` and `Parameter2`, `Nk` should go next to `scope` in `Parameter`.

Comment (PetarKirov) [reply spec/abi.dd:588]: (Because both `scope` and `return` can be combined with `out` and `ref`.)

Comment (kubo39) [reply spec/abi.dd:588]: Yes, I missed. Thanks for pointing out!

Comment (PetarKirov) [reply spec/abi.dd:588]: No worries, and excuse me for not being specific the first time I suggested adding `Nk` to the docs!

**Comment by PetarKirov** [comment]
I'll wait for the dmd pull request to be merged before merging this one.

**Pull request merged**